app/services/accept_npq_application.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+class AcceptNPQApplication
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  attribute :npq_application
+
+  validates :npq_application, presence: { message: I18n.t(:missing_npq_application) }
+  validate :not_already_accepted
+  validate :cannot_change_from_rejected
+  validate :other_accepted_applications_with_same_course?
+
+  def call
+    return self unless valid?
+
+    ApplicationRecord.transaction do
+      teacher_profile.update!(trn: npq_application.teacher_reference_number) if npq_application.teacher_reference_number_verified?
+      participant_profile
+      npq_application.update(lead_provider_approval_status: "accepted") && other_applications_in_same_cohort.update(lead_provider_approval_status: "rejected")
+      deduplicate_by_trn!
+      npq_application
+    end
+  end
+
+private
+
+  def not_already_accepted
+    return if npq_application.blank?
+
+    errors.add(:npq_application, I18n.t("activerecord.errors.models.npq_application.attributes.lead_provider_approval_status.has_already_been_accepted")) if npq_application.accepted?
+  end
+
+  def cannot_change_from_rejected
+    return if npq_application.blank?
+
+    errors.add(:npq_application, I18n.t("activerecord.errors.models.npq_application.attributes.lead_provider_approval_status.cannot_change_from_rejected")) if npq_application.rejected?
+  end
+
+  def cohort
+    npq_application.cohort
+  end
+
+  def other_accepted_applications_with_same_course?
+    errors.add(:npq_application, I18n.t("activerecord.errors.models.npq_application.attributes.lead_provider_approval_status.has_another_accepted_application")) if other_accepted_applications_with_same_course.present?
+  end
+
+  def other_accepted_applications_with_same_course
+    return if npq_application.blank?
+
+    @other_accepted_applications_with_same_course ||= NPQApplication
+                                                        .joins(:participant_identity)
+                                                        .where(participant_identity: { user_id: user.id })
+                                                        .where(npq_course: npq_course.rebranded_alternative_courses)
+                                                        .where(lead_provider_approval_status: "accepted")
+                                                        .where.not(id: npq_application.id)
+  end
+
+  def other_applications_in_same_cohort
+    @other_applications_in_same_cohort ||= NPQApplication
+      .joins(:participant_identity)
+      .where(participant_identity: { user_id: user.id })
+      .where(npq_course:)
+      .where(cohort:)
+      .where.not(id: npq_application.id)
+  end
+
+  def participant_profile
```

Comment:
could this be renamed to indicate a profile is being created?

Reply:
Hi @cwrw tks for your review. Addressed in https://github.com/DFE-Digital/early-careers-framework/pull/2674/commits/8c9fde545669f063b8a66cd2c524248a8025fdb5